web/js/chatRoom.jsx

```diff
@@ -0,0 +1,184 @@
+            {
+              messages.map(msg => {
+                return msg.from == user1 ? (
+                  <div key={msg.timestamp} className={`d-flex ${rightOrLeft(msg)} mb-4`}>
+                    <div className="img_cont_msg">
+                      <img src={`${image1_Or2(msg)}`} className="rounded-circle user_img_msg"/>
+                    </div>
+                    <div className="msg_cotainer">{msg.message}
+                      <span className="msg_time">{utils.formatDate(msg.timestamp, 'h:mm a, YYYY-MM-DD')}</span>
+                    </div>
+                  </div>
+                ) : 
+                  (
+                    <div key={msg.timestamp} className={`d-flex ${rightOrLeft(msg)} mb-4`}>
+                      <div className="msg_cotainer">{msg.message}
+                        <span className="msg_time">{utils.formatDate(msg.timestamp, 'h:mm a, YYYY-MM-DD')}</span>
+                      </div>
+                      <div className="img_cont_msg">
+                        <img src={`${image1_Or2(msg)}`} className="rounded-circle user_img_msg"/>
+                      </div>
+                    </div>
+                  )
+                ;
+              })
+            }
+
+
+            {/* <div className="d-flex justify-content-start mb-4">
+              <div className="img_cont_msg">
+                <img src={`${imgeurl}`} className="rounded-circle user_img_msg"/>
+              </div>
+              <div className="msg_cotainer">
+									Hi, how are you samim?
+                <span className="msg_time">8:40 AM, Today</span>
+              </div>
+            </div>
+            <div className="d-flex justify-content-end mb-4">
+              <div className="msg_cotainer_send">
+									Hi Khalid i am good tnx how about you?
+                <span className="msg_time_send">8:55 AM, Today</span>
+              </div>
+              <div className="img_cont_msg">
+                <img src={`${imgeurl2}`} className="rounded-circle user_img_msg"/>
+              </div>
+            </div> */}
+          </div>
+          <div className="card-footer"> 
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+}
+
+var ChatRoom = React.createClass({
+  contextTypes: {
+    router: React.PropTypes.object
+  },
+  
+  getInitialState: function () { 
+  
+    return {  
+      searchQuery: '',
+      results: null,
+      searchError: null, 
+      loading: false,
+      error: null
+    };
+  },
+
+  fetchChat: function(user1, user2) {
+
+    this.setState({ 
+      loading: true
+    });
+
+    utils.xhr({
+      method: 'get',
+      url: utils.apiPath('/users/chat/' + user1 + '/' +user2),
+    }, (err, res, data) => {
+      const error = err || (res.statusCode === 200 ? null : data);
+      this.setState({
+        loading: false,
+        error: error || null,
+        results: error ? null : data
+      }, () => {
+        if (this.state.searchError)
```

Comment:
It looks like `searchError` is never set anywhere. Indeed if I make the request fail (for instance when I had the `CHAT_ROOM_*` environment variable not configured correctly), the UI doesn't show any problem.

In case the request fails, the error should probably be set in `searchError`, right?

At the moment, errors are not being reported.